.github/workflows/linux.yml

```diff
@@ -18,7 +18,7 @@ jobs:
       matrix:
         # Use a tag from https://github.com/git/git/tags
         # Make sure to update `git-version` in the `run-tests` step as well.
-        git-version: ["v2.24.3", "v2.29.2", "v2.33.1", "v2.37.3"]
+        git-version: ["v2.24.3", "v2.29.2", "v2.33.1", "v2.37.3", "v2.45.1"]
```

Comment:
issue: It looks like https://github.com/git/git/commit/a8ae923f85da6434c3faf9c39719d6d5e5c77e65 is not actually included in v2.45.1, so this doesn't actually reproduce the issue 😅. So we can remove this for now.

That commit is on `next`, and it's causing tests to fail on the git-devel CI job in an incidental way: https://github.com/arxanas/git-branchless/actions/runs/9047534056/job/24859622426, though (not this bug as far as I can tell, but expected if the reference-transaction interface has changed).

If you can get a test case that reproduces this against git `next`, then we can include it in this PR to prevent regressions, a.s it should be checked by the daily "devel" job instead of regular PR checks.

Also, just to be sure, you might want to test with `cargo build --workspace && cargo test ...` due to [this attempted build-time optimization](https://github.com/arxanas/git-branchless/wiki/Runbook#testing-subcommands-in-isolation). In principle, the optimization shouldn't apply unless you opt-in, but it's possible there's a bug. Hooks, by design, involve calling a bunch of differences subprocesses, so I just wanted to raise it in case you see unexpected stale tests.